src/radical/pilot/agent/radical-pilot-agent-multicore.py

```diff
@@ -3242,7 +3247,8 @@ def stop(self):
         # During LRMS termination, we call any existing shutdown hooks on the
         # launch methods.  We only call LM shutdown hooks *once*
         launch_methods = set() # set keeps entries unique
-        launch_methods.add(self._cfg['mpi_launch_method'])
+        if 'mpi_launch_method' in self._cfg:
+            launch_methods.add(self._cfg['mpi_launch_method'])
         launch_methods.add(self._cfg['task_launch_method'])
```

Comment:
It is related, but changing the case for regular tasks here wouldn't make any difference because of other places in the code (both client and agent). So I suggest to leave it indeed.